core/shared/src/main/scala/scalaz/zio/internal/FiberContext.scala

```diff
@@ -374,6 +394,20 @@ private[zio] final class FiberContext[E, A](
 
   final def poll: UIO[Option[Exit[E, A]]] = ZIO.effectTotal(poll0)
 
+  final def inheritLocals: UIO[Unit] =
+    for {
+      locals <- copyFiberLocals
+      _ <- ZIO.foreach_(locals) {
```

Comment:
Good use of ZIO inside the interpreter to save code! `inheritLocals` doesn't need to be performance sensitive, I don't think.

I would add a short-circuit check, though. Maybe:

```scala
final def inheritLocals: UIO[Unit] = UIO.suspend {
  import scala.collection.JavaConverters._
  val locals = fiberRefLocals.asScala
  if (locals.size == 0) UIO.unit
  else UIO.foreach(locals) { ... }
}
```

This will ensure if there are no locals, it's as fast as it can possibly be.

Reply:
Changed, except that I use `locals.isEmpty` instead.